shell/components/Inactivity.vue

```diff
@@ -0,0 +1,208 @@
+<script>
+import ModalWithCard from '@shell/components/ModalWithCard';
+import { Banner } from '@components/Banner';
+import PercentageBar from '@shell/components/PercentageBar.vue';
+import throttle from 'lodash/throttle';
+import { MANAGEMENT } from '@shell/config/types';
+import { SETTING } from '@shell/config/settings';
+
+export default {
+  name:       'Inactivity',
+  components: {
+    ModalWithCard, Banner, PercentageBar
+  },
+  data() {
+    return {
+      enabled:             null,
+      isOpen:              false,
+      isInactive:          false,
+      showModalAfter:      null,
+      inactivityTimeoutId: null,
+      courtesyTimer:       null,
+      courtesyTimerId:     null,
+      courtesyCountdown:   null,
+      trackInactivity:     throttle(this._trackInactivity, 1000),
+    };
+  },
+  async mounted() {
+    const _settings = await this.$store.dispatch('management/find', { type: MANAGEMENT.SETTING, id: SETTING.UI_PERFORMANCE });
+
+    const settings = JSON.parse(_settings.value).inactivity;
```

Comment:
I get `expected end of JSON input` error.

Reply:
We should probably set a fallback and a chained condition for all the users which do not have any setting. I would recommend a unit test here, as it breaks the page.

Reply:
Yeah I've seen that too, but it did not work out 😬 
Looking forward to the updates 🍺